components/utils.py

```diff
@@ -1175,104 +1121,13 @@ def get(self, service_name):
-        rollback_props_path = self._get_rollback_props_file_path(
-            service_name)
-        if not os.path.isfile(rollback_props_path):
-            ctx.logger.debug('Archiving previous input properties for '
-                             'service {0}'.format(service_name))
-            mkdir(os.path.dirname(rollback_props_path))
-            properties_file_path = self._get_props_file_path(service_name)
-            move(properties_file_path, rollback_props_path)
-
-    def _get_props_file_path(self, service_name):
-        base_service_dir = self._get_properties_dir(service_name)
-        dest_file_path = os.path.join(base_service_dir,
-                                      self.PROPERTIES_FILE_NAME)
-        return dest_file_path
-
-    def _get_rollback_props_file_path(self, service_name):
-        base_service_dir = self.get_rollback_properties_dir(service_name)
-        dest_file_path = os.path.join(base_service_dir,
-                                      self.PROPERTIES_FILE_NAME)
-        return dest_file_path
-
-    def _load_ctx_properties(self, service_name):
-        node_props = ctx.node.properties.get_all()
-        # Use existing property configuration during upgrade
-        use_existing = node_props.get('use_existing_on_upgrade')
-        if use_existing:
-            existing_props = self.load_rollback_props(service_name)
-            # Removing properties with suffix matching upgrade properties
-            for key in existing_props.keys():
-                for suffix in self.UPGRADE_PROPS_SUFFIX:
-                    if key.endswith(suffix):
-                        del existing_props[key]
-
-            # Update node properties with existing configuration inputs
-            node_props.update(existing_props)
-
-        node_props['service_name'] = service_name
-        return node_props
-
-    def _get_properties_dir(self, service_name):
-        return os.path.join(self.BASE_PROPERTIES_PATH,
-                            service_name,
-                            self.NODE_PROPS_DIR_NAME)
-
-    def get_rollback_properties_dir(self, service_name):
-        return os.path.join(self.BASE_PROPERTIES_PATH,
-                            service_name,
-                            self.ROLLBACK_NODE_PROPS_DIR_NAME)
-
-    def _load_properties(self, service_name):
-        props_file = self._get_props_file_path(service_name)
-        with open(props_file) as f:
-            return json.load(f)
-
-    # This function should only be used when during upgrade workflow execution
-    def load_rollback_props(self, service_name):
-        upgrade_props_file = self._get_rollback_props_file_path(service_name)
-        if os.path.isfile(upgrade_props_file):
-            with open(upgrade_props_file) as f:
-                return json.load(f)
-        else:
-            ctx.logger.debug('Failed loading rollback properties. '
-                             'Properties file does not exist.')
+        return self._properties[service_name]
```

Comment:
remember that `.create()` and `.get()` are likely to be in different scripts, you need to still go through files, the instance won't just be sitting in memory for the whole duration of the bootstrap

Reply:
Oh, that's right. So I'd need to use a file